libc/string_lenchr.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX functions tests
+ * Tested functions:
+ * 	strlen()
+ * 	strnlen()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <limits.h>
+
+#include <unity_fixture.h>
+
+
+TEST_GROUP(string_strlen);
+TEST_GROUP(string_strnlen);
+
+
+TEST_SETUP(string_strlen)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strlen)
+{
+}
+
+
+TEST(string_strlen, strlen_simple)
+{
+
+	const char empty[] = "";
+	const char *msg = "Cristopher";
+	const char torn[] = "foo\0bar";
+
+	TEST_ASSERT_EQUAL_INT(8, strlen("New Case"));
+	TEST_ASSERT_EQUAL_INT(0, strlen(empty));
+	TEST_ASSERT_EQUAL_INT(10, strlen(msg));
+	TEST_ASSERT_EQUAL_INT(3, strlen(torn));
+	/*pangram with whole alphabet set*/
+	TEST_ASSERT_EQUAL_INT(43, strlen("The quick brown fox jumps over the lazy dog"));
+}
+
```

Comment:
Please create also the following test groups for strlen:

 - strlen_unprintable
 
 - strlen_ascii|